src/test/addrman_tests.cpp

```diff
@@ -52,6 +52,14 @@ class CAddrManTest : public CAddrMan
     {
         CAddrMan::Delete(nId);
     }
+
+    void SimConnFail(CService& addr)
```

Comment:
It's not clear to me what this function's name is an abbreviation of. "Simulate connection failure?" Is it for testing/debugging?

Reply:
Added comment

Reply:
But yes, it is for testing. The test before evict logic needs connections to fail to exercise the eviction code. This allows when we test a particular addresses to force it to fail.